pkg/agent/agentclient/client_test.go

```diff
@@ -0,0 +1,114 @@
+package agentclient
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"google.golang.org/grpc"
+	"sigs.k8s.io/apiserver-network-proxy/proto/agent"
+)
+
+func TestServeData_HTTP(t *testing.T) {
+	client := NewAgentClient("")
+	client.stream = newStream()
+	stopCh := make(chan struct{})
+
+	// Start agnet
+	go client.Serve(stopCh)
+	defer close(stopCh)
+
+	// Start test http server
+	expectedBody := "Hello, client"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, expectedBody)
+	}))
+	defer ts.Close()
+
+	// Send DIAL_REQ packet
```

Comment:
Simulate sending KAS DIAL_REQ from the Proxy to the AgentClient by sending appropriate packet